### PR TITLE
suit: Creation of separate .zip file for recovery application

### PR DESCRIPTION
**Pull request by ahasztag**
This commit introduces changes allowing for an automatic creation of a separate .zip file for DFU when buildin an application together with a SUIT recovery application.

### CODE REVIEW

**Review by tomchy — APPROVED** [review]

**Review by robertstypa — APPROVED** [review]

File: cmake/sysbuild/suit.cmake [hunk 1]
```diff
@@ -231,6 +253,14 @@ function(suit_create_package)
       )
     endif()
     suit_copy_artifact_to_output_directory(${image} ${BINARY_DIR}/zephyr/${BINARY_FILE})
+
+    unset(CONFIG_SUIT_RECOVERY)
+    sysbuild_get(CONFIG_SUIT_RECOVERY IMAGE ${image} VAR CONFIG_SUIT_RECOVERY KCONFIG)
+    if (CONFIG_SUIT_RECOVERY)
```
Comment (nordicjm) [inline cmake/sysbuild/suit.cmake:259]:
```suggestion
    if(CONFIG_SUIT_RECOVERY)
```
No spaces between if and bracket in cmake

File: cmake/sysbuild/suit.cmake [hunk 2]
```diff
@@ -272,6 +302,15 @@ function(suit_create_package)
 
     suit_render_template(${INPUT_ENVELOPE_JINJA_FILE} ${ENVELOPE_YAML_FILE} "${TEMPLATE_ARGS}")
     suit_create_envelope(${ENVELOPE_YAML_FILE} ${ENVELOPE_SUIT_FILE} ${SB_CONFIG_SUIT_ENVELOPE_SIGN})
+
+    unset(CONFIG_SUIT_RECOVERY)
+    sysbuild_get(CONFIG_SUIT_RECOVERY IMAGE ${image} VAR CONFIG_SUIT_RECOVERY KCONFIG)
+    if (CONFIG_SUIT_RECOVERY)
```
Comment (nordicjm) [inline cmake/sysbuild/suit.cmake:308]: here and etc.

**Review by nordicjm — APPROVED** [review]
Minor nit. (please fit when manifest is updated), build system changes OK

**Review by adsz-nordic — APPROVED** [review]

File: cmake/sysbuild/suit.cmake (hunk 1 above)
Comment (ahasztag) [reply cmake/sysbuild/suit.cmake]: Ok fixed

File: cmake/sysbuild/suit.cmake (hunk 2 above)
Comment (ahasztag) [reply cmake/sysbuild/suit.cmake]: Fixed everywhere in this file

**Pull request merged**